### DISCUSSION TITLE
My app suddenly missing

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#3263] [Help]
Hi, 

I'm using budibase self hosted and start create new app from scratch. After success created multi database table design and form then suddenly I got  error prompt message when try to revert form changes and my project missing from dashboard.

**Comment by nnasoke** [comment]
Same here. I used budibase's cloud, my project has all gone when I once try to click revert button and some error message pops up.

**Comment by mjashanks** [comment]
Thanks for reporting this. The "Revert" button is meant to return your app to the last published version. If you have never published, then the app gets removed. 

We should provide more protection against this - sorry. We will look at adding a simple check-in here. In future, we will be adding a more complete revision history, but that's a few months away, at least.

Sorry for this, I'm sure it could be a massive pain if you've encountered it.
